dispatches/models/fossil_case/properties/hitecsalt_properties.py

```diff
@@ -0,0 +1,424 @@
+
+            flags = {"Fflag": Fflag, "Tflag": Tflag, "Pflag": Pflag}
+
+        else:
+            # Check when the state vars are fixed already result in dof 0
+            for k in blk.keys():
+                if degrees_of_freedom(blk[k]) != 0:
+                    raise Exception("State vars fixed but degrees of freedom "
+                                    "for state block is not zero during "
+                                    "initialization.")
+
+        # Set solver options
+        if outlvl > 1:
+            stee = True
+        else:
+            stee = False
+
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Solve property correlation
+        results = solve_indexed_blocks(opt, [blk], tee=stee)
+
+        if outlvl > 0:
+            if results.solver.termination_condition \
+                    == TerminationCondition.optimal:
+                _log.info('{} Initialisation Step 1 Complete.'
+                          .format(blk.name))
+            else:
+                _log.warning('{} Initialisation Step 1 Failed.'
+                             .format(blk.name))
+
+        # ---------------------------------------------------------------------
+        if state_vars_fixed is False:
+            # release state vars fixed during initialization if control
+            # volume didn't fix the state vars
+            if hold_state is True:
+                return flags
+            else:
+                blk.release_state(flags)
+
+        if outlvl > 0:
+            if outlvl > 0:
+                _log.info('{} Initialisation Complete.'.format(blk.name))
+
+    def release_state(blk, flags, outlvl=0):
+        # Method to release states only if explicitly called
+
+        if flags is None:
+            return
+
+        # Unfix state variables
+        for k in blk.keys():
+            if flags['Fflag'][k] is False:
+                blk[k].flow_mass.unfix()
+            if flags['Pflag'][k] is False:
+                blk[k].pressure.unfix()
+            if flags['Tflag'][k] is False:
+                blk[k].temperature.unfix()
+
+        if outlvl > 0:
+            if outlvl > 0:
+                _log.info('{} State Released.'.format(blk.name))
+
+@declare_process_block_class("HitecsaltStateBlock", block_class=_StateBlock)
+class StateTestBlockData(StateBlockData):
+    """An example property package for Molten Salt properties."""
+
+    def build(self):
+        """Callable method for Block construction."""
+        super(StateTestBlockData, self).build()
+        self._make_params()
+        self._make_state_vars()
+        self._make_prop_vars()
+        self._make_constraints()
+
+    def _make_params(self):
+        """Make references to the necessary parameters contained."""
+        # Thermodynamic reference state
+        add_object_reference(self, "ref_temperature",
```

Comment:
Don't create reference for these. You can just use `self.params.parameter` to access these.

Reply:
addressed